src/thread_task.rs

```diff
@@ -514,16 +536,15 @@ pub(crate) unsafe fn rav1d_task_delayed_fg(
     in_0: *const Rav1dPicture,
 ) {
     let ttd: *mut TaskThreadData = &mut (*c).task_thread;
```

Comment:
You should make this a ref first, since mutability isn't being checked by the compiler right now.